custom_components/tuya_local/devices/thermoval_tvt40_thermostat.yaml

```diff
@@ -0,0 +1,315 @@
+name: Thermoval WiFi Thermostat
+products:
+  - id: jIRwsT5yLPJO41Po
+    name: TVT 40
+primary_entity:
+  entity: climate
+  translation_key: thermostat
+  dps:
+    - id: 2
+      name: temperature
+      type: integer
+      range:
+        min: 50
+        max: 400
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: Smart
+          value: program
+          icon: "mdi:home-thermometer"
+        - dps_val: Anti_frozen
+          value: away
+          icon: "mdi:snowflake-melt"
+        - dps_val: Manual
+          value: manual
+          icon: "mdi:cursor-pointer"
+    - id: 4
+      name: fault

```

Comment:
I would call this fault_code, and add a binary_sensor for it as well, that maps 0 to false, and any other value to true.  The binary sensor should be more convenient to use in automations and for an indication in the UI that there is a fault, and the attribute here gives access to the full fault code for reference.